storage/innobase/handler/ha_innodb.cc

```diff
@@ -18914,19 +18921,18 @@ wsrep_abort_transaction(
 	      wsrep_thd_transaction_state_str(victim_thd),
 	      wsrep_thd_query(victim_thd),
 	      victim_trx ? victim_trx->id : 0);
-
   if (victim_trx)
   {
+    wsrep_thd_UNLOCK(victim_thd);
```

Comment:
Potential race condition here: LOCK_thd_data is temporarily released, is it possible that the victim terminates the transaction at this point and victim_trx becomes invalid?

Reply:
Even worse, I noticed that `wsrep_thd_UNLOCK()` releases ALSO `LOCK_thd_kill`, whole `victim_thd` might be gone after this.

Reply:
As `wsrep_thd_UNLOCK()` releases victim `LOCK_thd_kill`, `find_thread_by_id()` must be used again after `lock_mutex_enter()` to get the `victim_thd` pointer.